src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -236,6 +317,241 @@ protected static async Task<ImmutableArray<SyntaxToken>> GetIdentifierOrGlobalNa
             };
         }
 
+        /// <summary>
+        /// Find references to a symbol inside global suppressions.
+        /// For example, consider a field 'Field' defined inside a type 'C'.
+        /// This field's documentation comment ID is 'F:C.Field'
+        /// A reference to this field inside a global suppression would be as following:
+        ///     [assembly: SuppressMessage("RuleCategory", "RuleId', Scope = "member", Target = "~F:C.Field")]
+        /// </summary>
+        protected static async Task<ImmutableArray<FinderLocation>> FindReferencesInDocumentInsideGlobalSuppressionsAsync(
```

Comment:
@CyrusNajmabadi The refactoring you suggested helped simplify the code quite a bit. This is still a big chunk of code, let me know if you prefer to move this method and its helpers into a separate partial declaration.

Reply:
yes.  please take this all to anotehr partial (or helper type).  thanks!  AbstractReferenceFinder_GlobalSuppressions.cs or something like that would be fine.